mountpoint-s3-client/src/mock_client.rs

```diff
@@ -493,6 +493,46 @@ impl ObjectClient for MockClient {
     }
 }
 
+#[derive(Debug)]
+pub struct MockPutObjectRequest {
+    key: String,
+    buffer: Vec<u8>,
+    objects: Arc<RwLock<BTreeMap<String, Arc<MockObject>>>>,
+    in_progress_uploads: Arc<RwLock<BTreeSet<String>>>,
+}
+
+impl MockPutObjectRequest {
+    fn new(
+        key: &str,
+        objects: &Arc<RwLock<BTreeMap<String, Arc<MockObject>>>>,
+        in_progress_uploads: &Arc<RwLock<BTreeSet<String>>>,
+    ) -> Self {
+        in_progress_uploads.write().unwrap().insert(key.to_owned());
```

Comment:
We don't rely on it anywhere, but this need a `Drop` impl to clean up if the caller didn't `complete`.